core/modules/system/system.module

```diff
@@ -1121,6 +1121,46 @@ function _system_batch_theme() {
   }
 }
 
+/**
+ * Helper function to deterine the MySQL or MariaDB versionl
+ *
+ * @see system_requirements().
+ * @see telemetry_telemetry_data().
+ */
+function _system_get_database_version() {
+  $version = Database::getConnection()->databaseType() === 'mysql' ? Database::getConnection()->version() : NULL;
+  // Backdrop core only supports MySQL and MariaDB. If there is no "maria"
+  // in the version string, assume it is MySQL.
+  $type = strpos(strtolower($version), 'maria') !== FALSE ? 'MariaDB' : 'MySQL';
+  // MariaDB version strings tend to have 2 versions, one for the equivalent
+  // MySQL version + another for the actual MariaDB version. For example, in
+  // 5.5.5-10.5.11-MariaDB:
+  // - 5.5.5 would denote the MySQL equivalency
+  // - 10.5.11 would be the actual MariaDB version.
+  // Grab all strings within the version that are a sequence of numbers and
+  // dots.
+  preg_match_all('/[\d\.]+/', $version, $versions);
+  if (count($versions[0]) == 1) {
+    $version = $versions[0][0];
+  }
+  else {
+    // Some variations of the MariaDB version string contain the version
+    // number more than once. For example:
+    // 5.5.5-10.5.11-MariaDB-1:10.5.11+maria~focal-log.
+    $versions = array_unique($versions[0]);
+    // Sort versions in reverse (descending) order.
+    rsort($versions, SORT_NATURAL);
+    // Get the greatest version (or the only version).
+    $version = $versions[0];
+  }
+  // Trim the version number to only major + minor, otherwise we'd flood the
+  // table with dozens of bug fix versions.
+  $version = explode(".", $version);
+  $version = array_slice($version, 0, 2);
```

Comment:
Since this version number is going to be used in multiple places, where in some of them the bug version may be important, we shouldn't be doing this here. Perhaps this function should be returning an array of separate type and version number, and then we can be doing the concatenation and trimming of the less significant version numbers only where appropriate. For example, use only 2 digits in telemetry, but the entire version number in the site status report.

Reply:
...here's a PR against this: https://github.com/jenlampton/backdrop/pull/17